cmd/promtool/tsdb.go

```diff
@@ -184,24 +91,24 @@ func (b *writeBenchmark) run() error {
 	if err != nil {
 		return err
 	}
+	st.DisableCompactions()
```

Comment:
Let's look on the race, if it's happening on prod code as well. Otherwise disable compaction sounds good here as this benchmarks ingestion only (:

Reply:
There is a race, also mentioned in #7643. But seems disabling compaction is not the correct way to solve this. I will update this pr after this problem is fixed.